File: server/src/main/java/fr/bookhub/entity/Loan.java
package fr.bookhub.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

import fr.bookhub.entity.Book;

@Entity
@Table(name = "loans")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Loan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @NotNull
    private LocalDateTime debutDate;
    @NotNull
    private LocalDateTime endDate;
    private LocalDateTime returnDate;
    @NotNull
    private Status status;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @NotNull
    private User user;

    @ManyToOne
    @NotNull
    private Book book;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "updated_by")
    private User updatedBy;
}

File: web/src/app/core/type/loan.ts
export interface Loan {
  id: number;
  debutDate: string;
  endDate: string;
  returnDate?: string | null;
  status: LoanStatus;
  late: boolean;
  userId: number;
  bookId: number;
  bookTitle?: string;
  bookAuthor?: string;
  bookCoverUrl?: string;
  createdAt?: string;
  updatedAt?: string;
  updatedBy?: number;
}

export type LoanStatus = 'IN_PROGRESS' | 'WAITING' | 'FINISHED' | 'LATE';


